src/Symfony/Component/Form/Extension/Core/DataTransformer/DateTimeToArrayTransformer.php

```diff
@@ -99,6 +99,8 @@ public function transform($dateTime)
                 // remove leading zeros
                 $entry = (string) (int) $entry;
             }
+            // unset reference to keep scope clear
+            unset($entry);
```

Comment:
@stof : this has sense only when foreach value variable is a reference. 
As function grows, variable (reference) got reused and "ghost" array change bug might happen.